modern/src/maki-interpreter/interpreter.js

```diff
@@ -154,7 +155,12 @@ export function* interpret(start, program, stack = []) {
           methodArgs.push(aValue);
         }
         const obj = popStackValue();
-        stack.push(obj[methodName](...methodArgs));
+        const ret = obj[methodName](...methodArgs);
+        if (ret && isPromise(ret)) {
+          stack.push(yield ret);
```

Comment:
Does this break the debugger? If so, we can fix in a subsequent PR.

Reply:
Yea, I think it would break the debugger right now